### ISSUE TITLE
Retire IQueryLanguage

### ISSUE THREAD

**Issue by danielcweber** [dotnet/reactive#526]
The IQueryLanguage interface was probably introduced to decouple the Observable-API (the Linq-operators) from the actual implementation. There is only one implementation of IQueryLanguage and I don't think that replacing functionality on different platforms was ever considered a feature (the interface is internal after all). Differences between platforms are handled by compiler directives, so I propose to get rid of the interface and inline the actual implementation.

**Comment by akarnokd** [comment]
I definitely want this. It is a pain to navigate from an operator in some code to the actual implementation of that operator, especially when the actual implementation is named differently or different overloads go to different implementations. I could do this once #536 is merged.

**Comment by akarnokd** [comment]
See #539 . I'll update #536 after it.

**Comment by bartdesmet** [comment]
See my remarks on https://github.com/dotnet/reactive/issues/540.

Rather than removing things to optimize the developer workflow of a handful of "Rx team" developers

> It is a pain to navigate from an operator in some code to the actual implementation of that operator

and for not quantitated performance arguments

> There's probably perf to be gained

it'd be great to make an effort to understand the original intent and consider the implications of breaking tools such as RxSpy. In fact, I'd advocate to leverage inherited assets as a starting point to build great tooling rather than ripping it out and bringing it back in another form later (unless good design arguments are made that an alternative approach is vastly superior).

**Comment by bartdesmet** [comment]
Closing this in favor of #540 where the approach on supporting debugging can be further discussed.
